sourcegraph/sourcegraph
# web: add previous url to `OpenTelemetry` spans

## Context

1. Split `TimeSinceNavigation` into `TimeSinceWindowLoad` and `TimeSincePageView`.
2. Add the previous URL to spans. Helpful in querying spans started upon leaving a specific part of the web application.
3. Add React component names to `react.component.name`.

## Test plan

1. `ENABLE_OPEN_TELEMETRY=true sg start web-standalone`
4. Open the browser console and ensure that debug-level output is enabled.
5. Open the search results page.
6. Click on the search result.
7. See spans created with the updated structure as described above.

<img width="579" alt="Screenshot 2022-11-12 at 16 24 09" src="https://user-images.githubusercontent.com/3846380/201465472-30bc9805-0a16-4c03-983f-08540d706d27.png">

## App preview:

- [Web](https://sg-web-vb-add-previous-url-to-spans.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
